### PR TITLE
updated UDC to also report store file sizes, in bytes

### PR DESCRIPTION
**Pull request by lassewesth**

### CODE REVIEW

File: community/udc/src/main/java/org/neo4j/ext/udc/impl/DefaultUdcInformationCollector.java [hunk 1]
```diff
@@ -150,10 +159,39 @@ public static String filterVersionForUDC( String version )
...
+    {
+        if (neoStoreDataSource == null) return;
+
+        try ( ResourceIterator<File> files = neoStoreDataSource.listStoreFiles( false ) )
+        {
+            addStoreFileSizes( udcFields, files );
+        }
+        catch ( IOException ignored )
+        {
+            // we just didn't report store file sizes this time around
```
Comment (davidegrohmann) [inline community/udc/src/main/java/org/neo4j/ext/udc/impl/DefaultUdcInformationCollector.java]: should we at least log something?

File: community/udc/src/main/java/org/neo4j/ext/udc/impl/DefaultUdcInformationCollector.java [hunk 2]
```diff
@@ -62,13 +37,41 @@
...
+import org.neo4j.kernel.NeoStoreDataSource;
+import org.neo4j.kernel.configuration.Config;
+import org.neo4j.kernel.impl.core.StartupStatistics;
+import org.neo4j.kernel.impl.store.id.IdGeneratorFactory;
+import org.neo4j.kernel.impl.store.id.IdType;
+import org.neo4j.kernel.impl.transaction.state.DataSourceManager;
+import org.neo4j.kernel.impl.util.OsBeanUtil;
+import org.neo4j.udc.UsageData;
+import org.neo4j.udc.UsageDataKeys;
+
```
Comment (davidegrohmann) [inline community/udc/src/main/java/org/neo4j/ext/udc/impl/DefaultUdcInformationCollector.java]: this is strange, we usually do not order imports this way

File: community/udc/src/main/java/org/neo4j/ext/udc/impl/DefaultUdcInformationCollector.java [hunk 3]
```diff
@@ -150,10 +159,39 @@ public static String filterVersionForUDC( String version )
...
+        }
+    }
+
+    private void addStoreFileSizes( Map<String, String> udcFields, ResourceIterator<File> files )
+    {
+        while ( files.hasNext() )
+        {
+            File file = files.next();
+
+            add( udcFields, STORE_FILE_SIZE + "_" +  file.getName().replaceAll( "\\.", "_" ), file.length() );
```
Comment (davidegrohmann) [inline community/udc/src/main/java/org/neo4j/ext/udc/impl/DefaultUdcInformationCollector.java]: maybe extract in a method the field name creation logic?

File: community/udc/src/test/java/org/neo4j/ext/udc/impl/CannedResourcesIterator.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.ext.udc.impl;
+
+import java.util.Collection;
+import java.util.Iterator;
+
+import org.neo4j.graphdb.ResourceIterator;
+
+class CannedResourcesIterator<T> implements ResourceIterator<T>
```
Comment (davidegrohmann) [inline community/udc/src/test/java/org/neo4j/ext/udc/impl/CannedResourcesIterator.java]: we can drop this, please use https://github.com/neo4j/neo4j/blob/3.1/community/collections/src/main/java/org/neo4j/helpers/collection/Iterators.java#L592 instead

File: community/udc/src/main/java/org/neo4j/ext/udc/impl/DefaultUdcInformationCollector.java (hunk 1 above)
Comment (lassewesth) [inline community/udc/src/main/java/org/neo4j/ext/udc/impl/DefaultUdcInformationCollector.java]: I don't think so, users would not care if their usage data wasn't sent - not their concern

File: community/udc/src/test/java/org/neo4j/ext/udc/impl/CannedResourcesIterator.java (hunk 4 above)
Comment (lassewesth) [inline community/udc/src/test/java/org/neo4j/ext/udc/impl/CannedResourcesIterator.java]: I was looking for that, thanks!

File: community/udc/src/main/java/org/neo4j/ext/udc/impl/DefaultUdcInformationCollector.java (hunk 3 above)
Comment (lassewesth) [inline community/udc/src/main/java/org/neo4j/ext/udc/impl/DefaultUdcInformationCollector.java]: Good idea.

**Pull request merged**